FWCore/Concurrency/interface/include_first_syncWait.h

```diff
@@ -8,32 +8,19 @@
 //  Created by Chris Jones on 2/24/21.
 //
 #include "FWCore/Concurrency/interface/WaitingTaskHolder.h"
+#include "FWCore/Concurrency/interface/FinalWaitingTask.h"
 #include "oneapi/tbb/task_group.h"
-#include "oneapi/tbb/task.h"
 #include <exception>
 
 namespace edm {
   template <typename F>
   [[nodiscard]] std::exception_ptr syncWait(F&& iFunc) {
     std::exception_ptr exceptPtr{};
-    //oneapi::tbb::task::suspend can only be run from within a task running in this arena. For 1 thread,
-    // it is often (always?) the case where not such task is being run here. Therefore we need
-    // to use a temp task_group to start up such a task.
     oneapi::tbb::task_group group;
-    group.run([&]() {
-      oneapi::tbb::task::suspend([&](oneapi::tbb::task::suspend_point tag) {
-        auto waitTask = make_waiting_task([tag, &exceptPtr](std::exception_ptr const* iExcept) {
-          if (iExcept) {
-            exceptPtr = *iExcept;
-          }
-          oneapi::tbb::task::resume(tag);
-        });
-        iFunc(WaitingTaskHolder(group, waitTask));
-      });  //suspend
-    });    //group.run
+    FinalWaitingTask last{group};
+    group.run([&]() { iFunc(WaitingTaskHolder(group, &last)); });  //group.run
```

Comment:
Does this case not need the `group.defer()`?

The workflow 4.68 step 2 fails with only
```
06-Dec-2022 23:15:05 CET  Initiating request to open file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/data/Run2012C/SingleElectron/USER/WElectron-PromptSkim-v3/000/199/812/00000/0286E6B0-35DB-E111-AF30-001A92810AA8.root
06-Dec-2022 23:15:09 CET  Successfully opened file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/data/Run2012C/SingleElectron/USER/WElectron-PromptSkim-v3/000/199/812/00000/0286E6B0-35DB-E111-AF30-001A92810AA8.root
```
in the log, which _might_ be a symptom of the job getting stuck?

Reply:
FinalWaitingTask has a defer builtin already.